src/csharp/Microsoft.Spark/RDD/Collector.cs

```diff
@@ -66,18 +65,10 @@ internal interface IDeserializer
         /// </summary>
         private sealed class BinaryDeserializer : IDeserializer
         {
-            [ThreadStatic]
-            private static MaxLengthReadStream s_slicedReadStream;
-
             private readonly BinaryFormatter _formater = new BinaryFormatter();
 
             public object Deserialize(Stream stream, int length)
             {
-                MaxLengthReadStream readStream = s_slicedReadStream ??
-                    (s_slicedReadStream = new MaxLengthReadStream());
-
-                readStream.Reset(stream, length);
-
                 return _formater.Deserialize(stream);
```

Comment:
important: I have removed the usage of `MaxLengthReadStream` and the type itself because it was not really used here.

```cs
_formater.Deserialize(stream); // stream istead of readStream is passed to the formatter
```

should it be removed? or changed to be actually used?

Reply:
I think this can be removed.